### PR TITLE
[FEATURE] Cleanup IndexUtility

### PR DESCRIPTION
**Pull request by NeoBlack**

### CODE REVIEW

File: Classes/Utility/IndexUtility.php [hunk 1]
```diff
@@ -50,11 +49,11 @@ public function __construct(
         string $indexName = null
     ) {
         $this->client = $client;
-        $this->configurationParser = $configurationParser;
         $this->logger = $logger;
         if (null !== $indexName) {
             $this->index = $this->client->getIndex($indexName);
         }
+        $this->configurationService = new ConfigurationService($client, $configurationParser, $logger);
```
Comment (psychomieze) [inline Classes/Utility/IndexUtility.php]: dependency injection, please... (see logger / parser / client)

File: Classes/Service/ConfigurationService.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+use Elastica\Client;
+use Elastica\Index;
+use Elastica\Type\Mapping;
+use Psr\Log\LoggerInterface;
+use T3G\Elasticorn\Utility\ConfigurationParser;
+use T3G\Elasticorn\Utility\DiffUtility;
+
+/**
+ * Class IndexUtility
```
Comment (psychomieze) [inline Classes/Service/ConfigurationService.php]: nope ;)

File: Classes/Utility/IndexUtility.php [hunk 3]
```diff
@@ -107,51 +106,42 @@ public function copyData(string $oldIndexName, string $newIndexName)
...
             $this->remap($indexName);
         }
     }
 
-
     /**
      * Creates configuration directories and files from settings and mappings of an existing index
      *
      * @param string $indexName
+     * @deprecated use ConfigurationService->createConfigurationFromExistingIndex
```
Comment (psychomieze) [inline Classes/Utility/IndexUtility.php]: wa? don't deprecate stuff, remove it please.

File: Classes/Utility/IndexUtility.php [hunk 4]
```diff
@@ -1,17 +1,16 @@
...
-use Elastica\Type\Mapping;
 use Psr\Log\LoggerInterface;
+use T3G\Elasticorn\Service\ConfigurationService;
 
 /**
  * Class IndexUtility
  *
- * @package T3G\Elasticorn
  */
 class IndexUtility
```
Comment (psychomieze) [inline Classes/Utility/IndexUtility.php]: Shouldn't this be a service, too?

Comment (NeoBlack) [inline Classes/Utility/IndexUtility.php]: yes could be, maybe in second patch?

File: Classes/Utility/IndexUtility.php (hunk 1 above)
Comment (NeoBlack) [inline Classes/Utility/IndexUtility.php]: yes, will push an update

**Comment by NeoBlack** [comment]
Tests are broken atm

**Comment by psychomieze** [comment]
Merged via #8 with fixed tests.

**Pull request closed without merging**